web/src/documents.ts

```diff
@@ -210,14 +215,43 @@ export async function cancel(
 }
 
 /**
- * Restart the document's kernel space
+ * Restart one, or all, of the kernels in a document's kernel space
+ *
+ * If `kernelId` is not supplied then all kernels in the kernel space
+ * will be restarted.
  */
 export async function restart(
   client: Client,
-  documentId: DocumentId
+  documentId: DocumentId,
+  kernelId?: KernelId
 ): Promise<void> {
   return client.call('documents.restart', {
     documentId,
+    kernelId,
+  }) as Promise<void>
+}
+
+/**
+ * Get a list of kernels in a document's kernel space
+ */
+export async function kernels(
+  client: Client,
+  documentId: DocumentId
+): Promise<void> {
+  return client.call('documents.kernels', {
+    documentId,
+  }) as Promise<void>
+}
+
+/**
+ * Get a list of symbols in a document's kernel space
+ */
+export async function symbols(
+  client: Client,
+  documentId: DocumentId
+): Promise<void> {
+  return client.call('documents.symbols', {
+    documentId,
   }) as Promise<void>
 }
```

Comment:
@alex-ketch I've added these two functions one to get a `Record<KernelId, KernelInfo>` and the other to get a `Record<String, SymbolInfo>`. At present there is a TypeScript type for `SymbolInfo` but not for `KernelInfo`. Its Rust is:

```rust

/// An identifier for a kernel
///
/// This is *not* a UUID but rather a id that is unique to a
/// local kernel space. This allows more useful ids to be assigned
/// e.g. `python`, `r` etc.
pub type KernelId = String;

/// The status of a running kernel
#[derive(Debug, PartialEq, Clone, Serialize, Display)]
#[allow(dead_code)]
pub enum KernelStatus {
    Pending,
    Starting,
    Idle,
    Busy,
    Unresponsive,
    Stopping,
    Finished,
    Failed,
    Unknown,
}

/// Information on a running kernel
///
/// Used when displaying information to the user about currently
/// running kernels.
#[derive(Debug, Clone, Serialize)]
pub struct KernelInfo {
    /// The id of the kernel instance
    pub id: KernelId,

    /// The status of the kernel
    pub status: KernelStatus,

    /// The kernel spec
    #[serde(flatten)]
    pub spec: Kernel,

    /// Whether the kernel is interruptable on the current machine
    pub interruptable: bool,

    /// Whether the kernel is forkable on the current machine
    pub forkable: bool,
}
```